### PR TITLE
Bpe

### PR DESCRIPTION
**Pull request by ElanaPearl**
* Added support for BPE tokenizers (both trained on-the-fly and pre-trained)
* Moved MLM training script to use the utils instead of duplicating the RawTextDataset function (first use case of the chemberta imports!! woo!)
* Fixed import issue in utils/raw_text_dataset.py

### CODE REVIEW

**Comment by seyonechithrananda** [comment]
Looks great!

File: chemberta/masked-lm/train_roberta_mlm.py [hunk 1]
```diff
@@ -62,20 +72,31 @@ def main(argv):
...
+    elif FLAGS.tokenizer_type.upper() == "BPE":
+        tokenizer_path = FLAGS.output_tokenizer_dir
+        if not os.path.isdir(tokenizer_path):
+            os.makedirs(tokenizer_path)
+        
+        tokenizer = ByteLevelBPETokenizer()
+        tokenizer.train(files=FLAGS.dataset_path, vocab_size=FLAGS.vocab_size, min_frequency=FLAGS.BPE_min_frequency, special_tokens=["<s>","<pad>","</s>","<unk>","<mask>"])
+        tokenizer.save_model(tokenizer_path)
+    else:
+        print("Please provide a tokenizer path if using the SMILES tokenizer")
```
Comment (seyonechithrananda) [inline chemberta/masked-lm/train_roberta_mlm.py:87]: this flow looks good, we assume smiles-tokenizer by default

File: chemberta/masked-lm/train_roberta_mlm.py [hunk 2]
```diff
@@ -15,11 +19,13 @@
 from transformers import RobertaTokenizerFast
 from transformers import RobertaForMaskedLM
 
-from data_utils import RawTextDataset
+from chemberta.utils.raw_text_dataset import RawTextDataset
```
Comment (seyonechithrananda) [inline chemberta/masked-lm/train_roberta_mlm.py:22]: yay!

**Review by seyonechithrananda — COMMENTED** [review]
Looks good, should we change default vocab to 1000? or keep it at 52,000 (for reproducibility) and just change when we need to

**Comment by ElanaPearl** [comment]
> Looks good, should we change default vocab to 1000? or keep it at 52,000 (for reproducibility) and just change when we need to

I'm good with either, I assume 52000 is too large for smiles but we should test it out and evaluate

**Comment by seyonechithrananda** [comment]
> > Looks good, should we change default vocab to 1000? or keep it at 52,000 (for reproducibility) and just change when we need to
> 
> I'm good with either, I assume 52000 is too large for smiles but we should test it out and evaluate

Sounds good, I think we should change to 500-1000 since we were using 52,000 for 1M/10M pre-training and the vocab had some pretty irrelevant tokens because of it
